File: Hash Table/242. Valid Anagram/solution.swift
class Solution {
    func isAnagram(_ s: String, _ t: String) -> Bool  {
        
        let s_count = self.get_frequency(s: s)
        let t_count = self.get_frequency(s: t)
        
        return s_count == t_count
    }
    
    func get_frequency(s:String) -> [Character:Int] {
        
        var frequency : [Character: Int] = [:]
        
        for c in Array(s){
            frequency[c, default : 0] += 1
        }
        
        return frequency
    }
}



let s = "anagram"
let t = "nagaram"
print(Solution().isAnagram(s, t))

File: Hash Table/242. Valid Anagram/solution.py

class Solution:
    def isAnagram(self, s: str, t: str) -> bool:

        return self.get_frequency(s) == self.get_frequency(t)
    
    def get_frequency(self, s:str):

        frequency = {}

        for c in s:
            frequency[c] = frequency.get(c, 0)+1
        
        return frequency
        



if __name__ == '__main__':

    """ Example 1: """
    s = "anagram"
    t = "nagaram"
    print(Solution().isAnagram(s, t))